### DISCUSSION TITLE
Query SQL

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#267] [Q&A] [answered]
Dear Team Algoritma

Mohon arahannya, saya punya tabel `CXREF` dimana jenis rekening yang ada didalam tabel `CXREF` tersebut ada 2 type rekening, yaitu A dan B. Saya mau menambahkan informasi status rekening dimana informasi status rekening A berada di tabel DBAL dan B berada di tabel TBAL masing-masing berada di tabel yang berbeda-beda.

Bagaimana saya menggunakan syntax di SQL nya ya ? sementara syntax yang saya tahu hanyalah menggunakan left join..apakah bisa menggunakan syntax CASE WHEN dan bagaimana pengaplikasiannya ya ?

Berikut adalah script yang masih pada proses left join saja:

SELECT * FROM VISIONN.CXREF A
LEFT JOIN VISIONN.DBAL B ON B.DMACCT = A.CXNOAC
LEFT JOIN VISIONN.TBAL C ON C.TCDNO = A.CXNOAC

<img width="302" alt="image" src="https://user-images.githubusercontent.com/103427989/183831847-5e6e64da-362b-4725-a46b-3eb78b958863.png">

mohon saran dan petunjuknya,

Best regards,
JT

**Comment by rizkiajimahardika** [comment, marked as answer]
Dear @joejnt 
Untuk menghubungkan antara tabel satu dengan tabel lainnya, dapat menggunakan join. Lalu untuk CASE WHEN digunakan ketika dalam suatu kondisi tertentu.

Sekian penjelasan yang dapat disampaikan, semoga membantu.

**Comment by shinddayo** [comment]
Kalo diliat kan 1 type bisa macem2 statusnya ya. Karena key-nya sudah pake unique ID nya (CXNOAC,DMACCT dan  TCDNO) saya pikir tidak perlu liat type A dan B nya langsung aja kondisiin dia ada di tbl yang mana statusnya. mungkin seperti ini

SELECT A.* , CASE WHEN B.STATUS IS NULL THEN C.STATUS ELSE B.STATUS END STATUS FROM VISIONN.CXREF A
LEFT JOIN VISIONN.DBAL B ON A.CXNOAC= B.DMACCT 
LEFT JOIN VISIONN.TBAL C ON A.CXNOAC = C.TCDNO

kecuali 1 ID dimungkinkan bisa Type A dan B, mungkin bisa seperti ini

SELECT DISTINCT A.* 
, CASE 
WHEN A.TYPE='A' THEN B.STATUS 
WHEN A.TYPE='B' THEN C.STATUS
ELSE 'TIPE APA INI?' 
END STATUS 
FROM VISIONN.CXREF A
LEFT JOIN VISIONN.DBAL B ON A.CXNOAC= B.DMACCT 
LEFT JOIN VISIONN.TBAL C ON A.CXNOAC = C.TCDNO

**Reply by joejnt** [reply]
@shinddayo 

Terima kasih untuk sarannya, mohon arahannya apakah perbedaan syntax `SELECT * FROM` dengan `SELECT DISTINCT * FROM`. apakah pada penerapannya nanti tidak ada perbedaan ?

Best regards,
JT

**Reply by shinddayo** [reply]
Distinct sepemahaman saya ambil unique,biar gak duplicate. Krna saya gak paham datanya mas @joejnt Dan asumsinya td 1 Id bisa lebih dari 1 type,asumsi sayabiar lebih aman pake distinct

https://www.w3schools.com/sql/sql_distinct.asp

**Reply by joejnt** [reply]
@shinddayo 

Well noted, terima kasih untuk _insight_ nya

Best regards,
JT
